code/modules/power/engines/singularity/singularity.dm

```diff
@@ -400,6 +407,21 @@
 			return 0
 	return 1
 
+// Returns a list of the field generators generating the containing field(if there is one within 10 tiles)
+/obj/singularity/proc/find_field_gens()
+	for(var/_dir in list(NORTH, SOUTH, EAST, WEST))
+		var/turf/T = loc
+		for(var/i in 1 to 10)
+			T = get_step(T, _dir)
+			var/obj/generator_field
```

Comment:
Just set this type to be `/obj/machinery/field/containment`, and then change the `if(generator_field)`s to `if(istype(generator_field))`

Reply:
I would still need to look for one on the turf, no? What do I gain from this?

Reply:
Oh, I see. I missed the second check, sorry. Do you think you could rename the variable generator_field to something a bit more generic, then?

Reply:
Done